src/handlers/next/process-saleor-webhook.ts

```diff
@@ -0,0 +1,121 @@
+import { NextApiRequest } from "next";
+import getRawBody from "raw-body";
+
+import { APL } from "../../APL";
+import { AuthData } from "../../APL/apl";
+import { getSaleorHeaders } from "../../headers";
+import { verifySignature } from "../../verify-signature";
+import { getBaseURL } from "./get-base-url";
+
+// TODO: Resolve shadow lint issue
+// eslint-disable-next-line no-shadow
+export enum SaleorWebhookError {
+  MISSING_HOST_HEADER,
+  MISSING_DOMAIN_HEADER,
+  MISSING_EVENT_HEADER,
+  MISSING_PAYLOAD_HEADER,
+  MISSING_SIGNATURE_HEADER,
+  MISSING_REQUEST_BODY,
+  WRONG_EVENT,
+  NOT_REGISTERED,
+  SIGNATURE_VERIFICATION_FAILED,
+  WRONG_METHOD,
+}
+
+export interface SaleorWebhookContext<T> {
+  data?: { baseUrl: string; event: string; payload: T; authData: AuthData };
+  error?: SaleorWebhookError;
```

Comment:
This should be typed with union, because its always *either* data *or* error

```
{
  data?: { baseUrl: string; event: string; payload: T; authData: AuthData };
} |
{
  error?: SaleorWebhookError;
}
```